sctp.c

```diff
@@ -422,21 +425,7 @@ static int janus_sctp_incoming_data(struct socket *sock, union sctp_sockstore ad
 void janus_sctp_send_data(janus_sctp_association *sctp, char *label, char *protocol, gboolean textdata, char *buf, int len) {
 	if(sctp == NULL)
 		return;
-	if(sctp->pending_messages != NULL && !g_queue_is_empty(sctp->pending_messages)) {
-		/* Messages waiting in the queue, send those first */
-		janus_sctp_pending_message *m = g_queue_peek_head(sctp->pending_messages);
-		while(m != NULL) {
-			int res = janus_sctp_send_text_or_binary(sctp, m->id, m->textdata, m->buf, m->len);
-			if(res == -2) {
-				JANUS_LOG(LOG_WARN, "[%"SCNu64"] Got EAGAIN when trying to resend pending message on channel %"SCNu16"\n",
-					sctp->handle_id, m->id);
-				break;
-			}
-			(void)g_queue_pop_head(sctp->pending_messages);
-			janus_sctp_pending_message_free(m);
-			m = g_queue_peek_head(sctp->pending_messages);
-		}
-	}
+	
```

Comment:
I'm not sure removing this is a good idea. The main reason why we tried to send all pending messages before sending the data we were just asked to send was to ensure the proper order of the messages. Since you removed this, and now pending messages are sent by the "dry" event, it means it's going to be very likely that a message sent now with `janus_sctp_send_data` will be sent _before_ a pending message waiting to be sent.

Reply:
No, this can not happen.
`janus_sctp_send_data` will either enqueue the message (in case we are buffering in `pending_message`) or send it right away.
Once `DRY_EVENT` occurs, the `pending_message` will be traversed in FIFO order.